tests/test_optimizer.py

```diff
@@ -546,6 +546,52 @@ def test_binary_annotation(self):
         self.assertEqual(expression.right.this.left.type.this, exp.DataType.Type.INT)
         self.assertEqual(expression.right.this.right.type.this, exp.DataType.Type.INT)
 
+    def test_interval_math_annotation(self):
+        schema = {
+            "x": {
+                "a": "DATE",
+                "b": "DATETIME",
+            }
+        }
+        for sql, expected_type, *expected_sql in [
+            (
+                "SELECT '2023-01-01' + INTERVAL '1' DAY",
+                exp.DataType.Type.DATE,
+                "SELECT CAST('2023-01-01' AS DATE) + INTERVAL '1' DAY",
+            ),
+            (
+                "SELECT '2023-01-01' + INTERVAL '1' HOUR",
+                exp.DataType.Type.DATETIME,
+                "SELECT CAST('2023-01-01' AS DATETIME) + INTERVAL '1' HOUR",
+            ),
+            (
+                "SELECT '2023-01-01 00:00:01' + INTERVAL '1' HOUR",
+                exp.DataType.Type.DATETIME,
+                "SELECT CAST('2023-01-01 00:00:01' AS DATETIME) + INTERVAL '1' HOUR",
+            ),
+            ("SELECT 'nonsense' + INTERVAL '1' DAY", exp.DataType.Type.UNKNOWN),
+            ("SELECT x.a + INTERVAL '1' DAY FROM x AS x", exp.DataType.Type.DATE),
+            ("SELECT x.a + INTERVAL '1' HOUR FROM x AS x", exp.DataType.Type.DATE),
```

Comment:
Why is this a date? This doesn't seem to match MySQL's behavior, for example:

```
mysql> select cast('2020-01-01' as date) + interval '1' hour;
+------------------------------------------------+
| cast('2020-01-01' as date) + interval '1' hour |
+------------------------------------------------+
| 2020-01-01 01:00:00                            |
+------------------------------------------------+
1 row in set (0.00 sec)
```

Reply:
Ah, you're right. This should be a datetime. good catch